beacon_node/beacon_chain/src/block_verification.rs

```diff
@@ -1702,6 +1702,9 @@ fn cheap_state_advance_to_obtain_committees<'a, E: EthSpec>(
     let block_epoch = block_slot.epoch(E::slots_per_epoch());
 
     if state.current_epoch() == block_epoch {
+        // Build both the current and previous epoch caches, as the previous epoch caches are
+        // useful for verifying attestations in blocks from the current epoch.
+        state.build_committee_cache(RelativeEpoch::Previous, spec)?;
```

Comment:
This is the essential bugfix IMO, even with this change alone the previous epoch cache errors disappear.

Forcing the caches to be built on the checkpoint state is good practice though, and contributes to improved performance.